### ISSUE TITLE
Receiving messages on wrong dispatcher

### ISSUE THREAD

**Issue by dpcleve** [AsynkronIT/protoactor-dotnet#482]
Hello,
I'm trying to use Proto.Actor (0.2.0) in a WPF application (targeting .net core 3.0).  I created a RootContext in the app startup and spawn an actor.  When the actor gets the Started message, it arrives on a thread pool thread (thread 4) as expected.  Then I spawn another actor ".WithDispatcher(new Proto.Mailbox.CurrentSynchronizationContextDispatcher())".  That actor receives Started on the UI thread (thread 1) as expected, then sends a message using "context.Send" to the first actor.  The first actor receives the message on the UI thread instead of thread pool.

When I receive the message and inspect the context in the debugger, it shows context._props.Dispatcher is ThreadPoolDispatcher, yet Thread.CurrentThread shows that we're in the main thread (thread 1)

Any idea why the first actor would receive on the UI thread, and not the thread pool?  I cannot seem to figure out how to make actors receive messages on the thread pool.  I've even tried, "WithDispatcher(new Proto.Mailbox.ThreadPoolDispatcher())", but that has no effect.

Basically, it seems any message sent from a "UI" actor always gets received on the UI regardless of the dispatcher the receiver was spawned with.

**Comment by sudsy** [comment]
I think the problem might relate to this line.

https://github.com/AsynkronIT/protoactor-dotnet/blob/0ff8eebaa75fc2b59236ff55e830efb07860da28/src/Proto.Mailbox/Dispatcher.cs#L50

I have experienced similar issues with a different use case and am currently investigating changing this line to Task.Run to ensure that it uses the ThreadPool rather than the current context.

See: https://devblogs.microsoft.com/pfxteam/task-run-vs-task-factory-startnew/

**Comment by dpcleve** [comment]
Thanks for the reply.  You are right, is it related to the provided dispatcher, but I'm not exactly sure why it doesn't work.  I was able to resolve this by writing my own UI dispatcher, which works as expected.

`public class UIDispatcher : IDispatcher
{
        public int Throughput => 300;
        public void Schedule(Func<Task> runner)
        {
            Application.Current.Dispatcher.InvokeAsync(runner);
        }
}`

**Comment by alexeyzimarev** [comment]
@dpcleve have you tried using `CurrentSynchronizationContextDispatcher`?

**Comment by dpcleve** [comment]
Yes.  But it did not work.  I think it may be related to .NET Core somehow.  I never did discover the root cause.  I wrote my own dispatcher, which worked, so I just moved on.  Then later, we had a problem with buttons not getting redrawn correctly.  We switched from .NET Core back to .NET framework 4.8 which resolved that issue, and so we abandoned WPF on .net core for now.

**Comment by rogeralsing** [comment]
`CurrentSynchronizationContextDispatcher` is the correct answer here. Lets file a bug report if this shows up again
